ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor.hs

```diff
@@ -561,8 +569,73 @@ $peer-churn-governor
 
 -- |
 --
-peerChurnGovernor :: MonadSTM m
-                  => PeerSelectionTargets
-                  -> m () --Void
-peerChurnGovernor _ =
-    return ()
+peerChurnGovernor :: forall m.
+                     ( MonadSTM m
+                     , MonadMonotonicTime m
+                     , MonadDelay m
+                     )
+                   => PeerSelectionTargets
+                 -> StrictTVar m PeerSelectionTargets
+                  -> m Void
+peerChurnGovernor base peerSelectionVar = do
+  -- Wait a while so that not only the closest peers have had the time
+  -- to become warm.
+  startTs0 <- getMonotonicTime
```

Comment:
TODO: revisit the policy once we have local root peers in the governor.  Intention is to give local root peers give head start and avoid giving advantage to hostile and quick root peers.